src/server/utils/dates.py

```diff
@@ -88,83 +89,56 @@ def dates_to_ranges(values: Optional[Sequence[Union[Tuple[int, int], int]]]) ->
-    intervals = []
-
-    # populate list of intervals based on original values
-    for v in values:
-        if isinstance(v, int):
-            # 202001 -> [202001, 202001]
-            intervals.append([week_value_to_week(v), week_value_to_week(v)])
-        else: # tuple
-            # (202001, 202002) -> [202001, 202002]
-            intervals.append([week_value_to_week(v[0]), week_value_to_week(v[1])])
-
-    intervals.sort(key=lambda x: x[0])
-
-    # merge overlapping intervals https://leetcode.com/problems/merge-intervals/
-    merged = []
-    for interval in intervals:
-        # no overlap, append the interval
-        # caveat: we subtract 1 from interval[0] so that contiguous intervals are considered overlapping. i.e. [1, 1], [2, 2] -> [1, 2]
-        if not merged or merged[-1][1] < interval[0] - 1:
-            merged.append(interval)
-        # overlap, merge the current and previous intervals
-        else:
-            merged[-1][1] = max(merged[-1][1], interval[1])
+    return _to_ranges(values, week_value_to_week, week_to_time_value, 1)
 
-    # convert intervals from weeks back to integers
-    ranges = []
-    for m in merged:
-        if m[0] == m[1]:
-            ranges.append(week_to_time_value(m[0]))
-        else:
-            ranges.append((week_to_time_value(m[0]), week_to_time_value(m[1])))
-
-    return ranges
+def _to_ranges(values: Sequence[Union[Tuple[int, int], int]], value_to_date: Callable, date_to_value: Callable, time_unit: Union[int, timedelta]) -> Sequence[Union[Tuple[int, int], int]]:
+    try:
+        intervals = []
+
+        # populate list of intervals based on original date/week values
+        for v in values:
+            if isinstance(v, int):
+                # 20200101 -> [20200101, 20200101]
+                intervals.append([value_to_date(v), value_to_date(v)])
+            else: # tuple
+                # (20200101, 20200102) -> [20200101, 20200102]
+                intervals.append([value_to_date(v[0]), value_to_date(v[1])])
+
+        intervals.sort()
+
+        # merge overlapping intervals https://leetcode.com/problems/merge-intervals/
+        merged = []
+        for interval in intervals:
+            # no overlap, append the interval
+            # caveat: we subtract 1 from interval[0] so that contiguous intervals are considered overlapping. i.e. [1, 1], [2, 2] -> [1, 2]
+            if not merged or merged[-1][1] < interval[0] - time_unit:
+                merged.append(interval)
+            # overlap, merge the current and previous intervals
+            else:
+                merged[-1][1] = max(merged[-1][1], interval[1])
+
+        # convert intervals from dates/weeks back to integers
+        ranges = []
+        for m in merged:
+            if m[0] == m[1]:
+                ranges.append(date_to_value(m[0]))
+            else:
+                ranges.append((date_to_value(m[0]), date_to_value(m[1])))
+
+        return ranges
+    except:
```

Comment:
a `logger.info` that reports the reason for this fall-through would be good.  itll almost certainly be a bad conversion in `value_to_date()`, in which case we might wanna see what bad inputs we are being handed.